Fails to build against Boost 1.67

Regressed by https://github.com/boostorg/uuid/commit/a8a1ec3d826b. Found [downstream](https://reviews.freebsd.org/D15030).  CC @yurivict

```c++
$ cmake -GNinja .
$ ninja
[...]
In file included from outofcore/tools/outofcore_print.cpp:53:
In file included from outofcore/include/pcl/outofcore/outofcore_impl.h:50:
outofcore/include/pcl/outofcore/impl/octree_disk_container.hpp:77:74: error: no matching constructor for initialization of 'boost::uuids::random_generator' (aka 'boost::uuids::random_generator_pure')
    boost::uuids::random_generator OutofcoreOctreeDiskContainer<PointT>::uuid_gen_ (&rand_gen_);
                                                                         ^          ~~~~~~~~~~
/usr/local/include/boost/uuid/random_generator.hpp:149:7: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'boost::mt19937 *' (aka 'mersenne_twister_engine<unsigned int, 32, 624, 397, 31, 2567483615U, 11, 4294967295U, 7, 2636928640U, 15, 4022730752U, 18, 1812433253> *') to 'const boost::uuids::random_generator_pure' for 1st argument
class random_generator_pure
      ^
/usr/local/include/boost/uuid/random_generator.hpp:149:7: note: candidate constructor (the implicit default constructor) not viable: requires 0 arguments, but 1 was provided
1 error generated.
```

Error logs:
- [FreeBSD 11.1 amd64](http://package22.nyi.freebsd.org/data/111amd64-default-PR227427/2018-04-17_17h22m51s/logs/errors/pcl-pointclouds-1.8.1_2.log)
- [FreeBSD 11.1 i386](http://package23.nyi.freebsd.org/data/111i386-default-PR227427/2018-04-18_09h58m25s/logs/errors/pcl-pointclouds-1.8.1_2.log)
- [FreeBSD 10.3 amd64](http://package22.nyi.freebsd.org/data/103amd64-default-PR227427/2018-04-18_08h05m01s/logs/errors/pcl-pointclouds-1.8.1_2.log)
- [FreeBSD 10.3 i386](http://package23.nyi.freebsd.org/data/103i386-default-PR227427/2018-04-18_08h05m08s/logs/errors/pcl-pointclouds-1.8.1_2.log)

pcl1.8 make error
[ 17%] Built target pcl_sample_consensus
[ 17%] Building CXX object io/CMakeFiles/pcl_io.dir/src/openni_camera/openni_image_rgb24.cpp.o
Scanning dependencies of target pcl_stereo
[ 17%] Building CXX object stereo/CMakeFiles/pcl_stereo.dir/src/stereo_grabber.cpp.o
[ 17%] Building CXX object io/CMakeFiles/pcl_io.dir/src/openni_grabber.cpp.o
/home/cc/software/pcl-1.8-master/stereo/src/stereo_grabber.cpp:44:32: internal compiler error: Segmentation fault
 struct pcl::StereoGrabberBase::StereoGrabberImpl
                                ^
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-5/README.Bugs> for instructions.
stereo/CMakeFiles/pcl_stereo.dir/build.make:62: recipe for target 'stereo/CMakeFiles/pcl_stereo.dir/src/stereo_grabber.cpp.o' failed
make[2]: *** [stereo/CMakeFiles/pcl_stereo.dir/src/stereo_grabber.cpp.o] Error 1
CMakeFiles/Makefile2:6068: recipe for target 'stereo/CMakeFiles/pcl_stereo.dir/all' failed
make[1]: *** [stereo/CMakeFiles/pcl_stereo.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 18%] Building CXX object io/CMakeFiles/pcl_io.dir/src/oni_grabber.cpp.o
[ 18%] Building CXX object io/CMakeFiles/pcl_io.dir/src/dinast_grabber.cpp.o
[ 18%] Building CXX object io/CMakeFiles/pcl_io.dir/src/libpng_wrapper.cpp.o
[ 18%] Linking CXX shared library ../lib/libpcl_io.so
[ 18%] Built target pcl_io
Makefile:160: recipe for target 'all' failed
make: *** [all] Error 2

--- a/outofcore/include/pcl/outofcore/impl/octree_disk_container.hpp
+++ b/outofcore/include/pcl/outofcore/impl/octree_disk_container.hpp
@@ -74,7 +74,7 @@ namespace pcl
     OutofcoreOctreeDiskContainer<PointT>::rand_gen_ (static_cast<unsigned int> (std::time(NULL)));
 
     template<typename PointT>
-    boost::uuids::random_generator OutofcoreOctreeDiskContainer<PointT>::uuid_gen_ (&rand_gen_);
+    boost::uuids::basic_random_generator<boost::mt19937> OutofcoreOctreeDiskContainer<PointT>::uuid_gen_ (&rand_gen_);
 
     template<typename PointT>
     const uint64_t OutofcoreOctreeDiskContainer<PointT>::READ_BLOCK_SIZE_ = static_cast<uint64_t> (2e12);

--- a/outofcore/include/pcl/outofcore/octree_disk_container.h
+++ b/outofcore/include/pcl/outofcore/octree_disk_container.h
@@ -296,7 +296,7 @@ namespace pcl
 
         static boost::mutex rng_mutex_;
         static boost::mt19937 rand_gen_;
-        static boost::uuids::random_generator uuid_gen_;
+        static boost::uuids::basic_random_generator<boost::mt19937> uuid_gen_;
 
     };
   } //namespace outofcore

